### PR TITLE
Iss166 locpol

### LINKED ISSUE #166: Allow estimate_npmetric_erf to avoid uncount steps
Load a new library

`library("locpol")
`
Change
`  erf <- stats::approx(KernSmooth::locpoly(matched_w, matched_Y, bandwidth=h_opt), xout=w_vals)$y
`
to
`erf <- locpol::locpol(formula = Y~w, data = data, bw = h_opt, weig = counter, xeval = a.w_vals, kernel = gaussK)`

This will allow us to avoid uncount steps in the function, and save memories. Also, it will make the non-parametric implementation more aligns with semi-/parametric implementations.

The function in https://github.com/fasrc/CausalGPS/blob/develop/R/smooth_erf.R will need to change accordingly as well.

### PR DESCRIPTION
**Pull request by Naeemkh**
Closes #166

### CODE REVIEW

File: R/compute_risk.R [hunk 1]
```diff
@@ -8,15 +8,19 @@
 #' @param h A scalar representing the bandwidth value.
 #' @param matched_Y A vector of outcome variable in the matched set.
 #' @param matched_w A vector of continuous exposure variable in the matched set.
+#' @param matched_cw A vector of counter or weight variable in the matched set.
```
Comment (wxwx1993) [inline R/compute_risk.R:11]: the variable names "matched_w" and "matched_cw" are too similar to each other, yet represent very different means. I would suggest just calling it "weights"

File: functional_tests/ft_estimate_npmetric_erf.R [hunk 2]
```diff
@@ -3,7 +3,7 @@ m_d <- generate_syn_data(sample_size = 2000)
 pseudo_pop <- generate_pseudo_pop(m_d$Y,
                                  m_d$treat,
                                  m_d[c("cf1","cf2","cf3","cf4","cf5","cf6")],
-                                 ci_appr = "matching",
+                                 ci_appr = "weighting",
```
Comment (wxwx1993) [inline functional_tests/ft_estimate_npmetric_erf.R:6]: Why do you change "matching" to "weighting" here?

Comment (Naeemkh) [reply functional_tests/ft_estimate_npmetric_erf.R:6]: ft_* are functional tests. It is just an example.

File: R/compute_risk.R (hunk 1 above)
Comment (Naeemkh) [reply R/compute_risk.R:11]: Should we completely drop `matched_` and go with `Y`, `w`, and `counter_weight`?

File: functional_tests/ft_estimate_npmetric_erf.R (hunk 2 above)
Comment (wxwx1993) [reply functional_tests/ft_estimate_npmetric_erf.R:6]: OK!

File: R/compute_risk.R (hunk 1 above)
Comment (wxwx1993) [reply R/compute_risk.R:11]: Yes, that is a great idea!

**Pull request merged**